app/ui-react/syndesis/src/modules/integrations/components/editor/choice/ChoiceStepPage.tsx

```diff
@@ -81,6 +81,11 @@ export class ChoiceStepPage extends React.Component<IChoiceStepPageProps> {
                   const onUpdatedIntegration = async (
                     values: IChoiceFormConfiguration
                   ) => {
+                    // this absolutely has to be set before we go
+                    // creating flows that need to point to it
+                    if (typeof state.step.id === 'undefined') {
```

Comment:
Better do this [here](https://github.com/syndesisio/syndesis-react/blob/1f2a94f31bd94a3a1e2968a2f33bc3e8feca9a23/app/ui-react/packages/api/src/useIntegrationHelpers.tsx#L29)

Reply:
so...  it's a real chicken and egg problem.  The step ID is needed to maintain it's association with flows.  I've changed this now to create an updated step object instead of mutating the state.